src/base/bittorrent/session.cpp

```diff
@@ -1347,6 +1361,48 @@ bool Session::addTorrent_impl(AddTorrentData addData, const MagnetUri &magnetUri
     return true;
 }
 
+bool Session::findIncompleteFiles(TorrentInfo &torrentInfo, QString &savePath) const
+{
+    auto findInDir = [](const QString &dirPath, TorrentInfo &torrentInfo) -> bool
+    {
+        bool found = false;
+        if (torrentInfo.filesCount() == 1) {
+            const QString filePath = dirPath + torrentInfo.filePath(0);
+            if (QFile(filePath).exists()) {
+                found = true;
+            }
+            else if (QFile(filePath + QB_EXT).exists()) {
+                found = true;
+                torrentInfo.renameFile(0, torrentInfo.filePath(0) + QB_EXT);
+            }
+        }
+        else {
+            QStringList allFiles =
+                    findAllFiles(dirPath + Utils::String::fromStdString(torrentInfo.nativeInfo()->files().name()));
+            for (int i = 0; i < torrentInfo.filesCount(); ++i) {
```

Comment:
>a) why does it continue looping if true was already set?

I've just explained!
>We have to process all files to be able to start with a partially downloaded torrent, when the "Append .!qB extension" option is enabled.

>b) what if the torrent first file is, say, "description.txt" or "info.nfo" or any other common name which is easy to find at a random place?

If the user tries to save the torrent in a directory that already contains files with the same names, but different content, it is the problem of the user. He must think too!.. Here I have not made anything, just leave everything as it was before.